app/src/main/kotlin/com/wire/android/ui/home/newconversation/NewConversationRouter.kt

```diff
@@ -18,55 +19,40 @@ import com.wire.android.R
 import com.wire.android.ui.common.topappbar.NavigationIconType
 import com.wire.android.ui.common.topappbar.WireCenterAlignedTopAppBar
 import com.wire.android.ui.common.topappbar.search.AppTopBarWithSearchBar
-import com.wire.android.ui.home.newconversation.NewConversationNavigationCommand.KnownContacts
-import com.wire.android.ui.home.newconversation.NewConversationNavigationCommand.SearchContacts
 import com.wire.android.ui.home.newconversation.contacts.ContactsScreen
+import com.wire.android.ui.home.newconversation.newGroup.NewGroupScreen
 import com.wire.android.ui.home.newconversation.search.SearchPeopleScreen
+import kotlinx.coroutines.flow.launchIn
+import kotlinx.coroutines.flow.onEach
 
 @OptIn(ExperimentalAnimationApi::class)
 @Composable
 fun NewConversationRouter(newConversationViewModel: NewConversationViewModel = hiltViewModel()) {
     val newConversationState = rememberNewConversationState()
+    val scope = rememberCoroutineScope()
 
-    LaunchedEffect(newConversationViewModel.navigateCommand) {
-        when (newConversationViewModel.navigateCommand) {
-            KnownContacts -> newConversationState.navigateToKnownContacts()
-            SearchContacts -> newConversationState.navigateToSearch()
-        }
+    LaunchedEffect(newConversationViewModel) {
```

Comment:
Why aren't we launching each effect upon any new `navigateCommand`? My understanding is that if we only use the `newConversationViewModel` as the key, this will be only launched once, on the VM initialisation?

Reply:
sorry  maybe I didn't get what do you mean correctly , but as far as I understand we did it this way so the VM can init once as it's shared with the 3 screens here (same flow) and the moveToStep will change it's value

Reply:
@MohammedMokresh  did you test if it makes any difference when you remove LaunchedEffect ? I think it shouldn't make a difference, but not sure